framework/decode/parsed_block.cpp

```diff
@@ -76,14 +76,18 @@ void ParsedBlock::Decompress(BlockParser& parser)
         {
             auto compressed_span     = GetCompressedSpan(args);
             auto uncompressed_size   = GetUncompressedSize(args);
-            auto uncompressed_buffer = parser.DecompressSpan(compressed_span, uncompressed_size);
-            // Patch the data buffer pointer, and shift ownership of the backing store to the parsed block
-            args.data = uncompressed_buffer.template GetAs<const uint8_t>();
-            UpdateUncompressedStore(std::move(uncompressed_buffer));
+            auto result              = parser.DecompressSpan(compressed_span, uncompressed_size);
+            if (result.success)
+            {
+                // Patch the data buffer pointer, and shift ownership of the backing store to the parsed block
+                args.data = result.decompressed_store.template GetAs<const uint8_t>();

```

Comment:
Won't this fail if `result.decompresssed_store` is empty (block_parser.cpp line 144)?

Reply:
decompress_store empty is correct when compressed store is empty GetAs will return a nullptr, which is correct for an empty args.data